src/openrct2/ride/TrackDesign.cpp

```diff
@@ -2109,20 +2110,20 @@ void track_design_draw_preview(TrackDesign* td6, uint8_t* pixels)
  * design preview.
  *  rct2: 0x006D1C68
  */
-static map_backup* track_design_preview_backup_map()
+static std::unique_ptr<map_backup> track_design_preview_backup_map()
 {
-    map_backup* backup = static_cast<map_backup*>(malloc(sizeof(map_backup)));
-    if (backup != nullptr)
-    {
-        std::memcpy(backup->tile_elements, gTileElements, sizeof(backup->tile_elements));
-        std::memcpy(backup->tile_pointers, gTileElementTilePointers, sizeof(backup->tile_pointers));
-        backup->next_free_tile_element = gNextFreeTileElement;
-        backup->map_size_units = gMapSizeUnits;
-        backup->map_size_units_minus_2 = gMapSizeMinus2;
-        backup->map_size = gMapSize;
-        backup->current_rotation = get_current_rotation();
-    }
-    return backup;
+    auto backup_ptr = std::make_unique<map_backup>();
+    if (backup_ptr.get() != nullptr)
```

Comment:
You only need to use get() when you absolutely need the pointer to the memory. None of these in this particular function require that. You can dereferrence a smart pointer as expected.